internal/services/keyvault/key_vault_resource.go

```diff
@@ -775,9 +775,12 @@ func resourceKeyVaultRead(d *pluginsdk.ResourceData, meta interface{}) error {
 	//
 	// We don't know if the private endpoint has been created yet, so we need
 	// to ignore the error if the data plane call fails.
+	//
+	// if the tenant id is not the same as the current account, we need to ignore the error
 	contacts, err := managementClient.GetCertificateContacts(ctx, vaultUri)
 	if err != nil {
-		if publicNetworkAccessEnabled && (!utils.ResponseWasForbidden(contacts.Response) && !utils.ResponseWasNotFound(contacts.Response)) {
+		theSameTenant := resp.Model == nil || resp.Model.Properties.TenantId == meta.(*clients.Client).Account.TenantId
+		if theSameTenant && publicNetworkAccessEnabled && (!utils.ResponseWasForbidden(contacts.Response) && !utils.ResponseWasNotFound(contacts.Response)) {
```

Comment:
This doesn't really solve the issue though, it just hides it?

Provided the Tenant ID is specified [in the `auxiliary_tenant_ids` field](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#auxiliary_tenant_ids) then we should be able to access this, and it should work - so I think that'd be the thing to do here rather than just ignoring the error.

Reply:
In which case we should be able to update that to handle it too? We've got the information needed to query it, so we should be able to request a token for the resource in question for that tenant?